microcosm_eventsource/immutable_event_factory.py

```diff
@@ -0,0 +1,92 @@
+"""
+Factory for immutable events.
+
+In extension of EventFactory, this class provides a native support of event updating container objects based
+on event types
+
+"""
+from typing import Dict
+
+from microcosm_eventsource.factory import EventFactory
+
+
+_common_container_mutator_registry: Dict[str, str] = {}
+_event_specific_container_mutator_registry: Dict[str, str] = {}
```

Comment:
I'm not too comfortable with those registries just being dictionaries defined in a module. It doesn't square with our use of dependency injection.
One thing you could do is instead put them in a graph component; then the decorators below would need to use the `on_resolve` hook to register the event types at graph load time. This is similar to what we do for the pubsub message schema registry (see [microcosm-pubsub](https://github.com/globality-corp/microcosm-pubsub/blob/develop/microcosm_pubsub/decorators.py#L15)).

Reply:
@afallou I understand what you are getting at here(regarding dependency injection).

If there were multiple components relying or accessing this registry, using DI would be right. In this case, these registries are private components of one class. The fact that their scope is not private, is something I can look at. Currently, the reason they are not within scope of ContainerMutatorEventFactory class, is to ease the usage of decorators. Otherwise, decorators would have to be ContainerMutatorEventFactory. common_container_mutator, which makes it harder to use.

Reply:
Right; using a component is not required here, but it is one solution to have the registry scope made private.